components/Feature/VulnerabilitiesGrid/index.js

```diff
@@ -1,19 +1,42 @@
-import { useState } from 'react';
+import { useState, useMemo } from 'react';
 import {
   Accordion,
   AccordionItem,
   Checkbox,
   CheckboxList,
   TextInput
 } from 'components/Form';
+import ResourceCard from './ResourceCard';
 import groups from './grid.json';
 
-const VulnerabilitiesGrid = ({ onUpdate }) => {
+function createLookup(resources) {
+  const lookup = new Map();
+
+  groups.forEach(group => {
+    const { name, assets, vulnerabilities } = group;
+    const values = new Set();
+
+    assets
+      .concat(vulnerabilities)
+      .map(item => item.label)
+      .concat(group.name)
+      .forEach(value => values.add(value));
+
+    lookup.set(name, values);
+  });
+
+  return lookup;
+}
+
+const VulnerabilitiesGrid = ({ resources, onUpdate }) => {
   const [grid, setGrid] = useState({ assets: {}, vulnerabilities: {} });
+  const groupItems = useMemo(() => createLookup(resources), resources);
```

Comment:
Wrapped in `useMemo` since this is pretty expensive to create on every single render.